uniqueidentifier.go

```diff
@@ -75,6 +76,25 @@ func (u UniqueIdentifier) String() string {
 
 // MarshalText converts Uniqueidentifier to bytes corresponding to the stringified hexadecimal representation of the Uniqueidentifier
 // e.g., "AAAAAAAA-AAAA-AAAA-AAAA-AAAAAAAAAAAA" -> [65 65 65 65 65 65 65 65 45 65 65 65 65 45 65 65 65 65 45 65 65 65 65 65 65 65 65 65 65 65 65]
-func (u UniqueIdentifier) MarshalText() []byte {
-	return []byte(u.String())
+func (u UniqueIdentifier) MarshalText() (text []byte, err error) {
+	text = []byte(u.String())
+	return
+}
+
+// Unmarshals a string representation of a UniqueIndentifier to bytes
+// "01234567-89AB-CDEF-0123-456789ABCDEF" -> [48, 49, 50, 51, 52, 53, 54, 55, 45, 56, 57, 65, 66, 45, 67, 68, 69, 70, 45, 48, 49, 50, 51, 45, 52, 53, 54, 55, 56, 57, 65, 66, 67, 68, 69, 70]
+func (u *UniqueIdentifier) UnmarshalJSON(b []byte) error {
+	input := string(b)
+	// remove quotes
+	input = strings.Trim(input, `"`)
+	// decode
+	bytes, err := hex.DecodeString(strings.ReplaceAll(input, "-", ""))
+
+	if err != nil {
+		return err
+	}
+	// Copy the bytes to the UniqueIdentifier
+	copy(u[:], bytes)
+
+	return nil
 }
```

Comment:
can this be moved to a uniqueidentifier_go112.go? Otherwise you need an older API than strings.ReplaceAll

Reply:
Just to make sure I'm on the same page, you mean I should move the newly added code to a new file called `uniqueidentifier_go112.go`?

Reply:
Assuming that this new interface is only needed for apps using a Go version newer than 1.11, then yes.
If you think apps using older Go versions need this implementation, you'll need to fix the build.

Currently 1.10 and 1.11 builds fail.

go build
# github.com/microsoft/go-mssqldb
.\uniqueidentifier.go:91:33: undefined: strings.ReplaceAll
Command exited with code 2

At some point we will upgrade the minimum supported go version for this driver.

Reply:
Ah I see. 
I've switched over to using `strings.Replace` instead and I see now that the go 1.10 and 1.11 build versions are passing